src/cbc_sdk/platform/alerts.py

```diff
@@ -997,6 +997,92 @@ def set_rows(self, rows):
         self._batch_size = rows
         return self
 
+    def set_time_range(self, *args, **kwargs):
+        """
+        Sets the 'time_range' query body parameter, determining a time window based on 'device_timestamp'.
+
+        Args:
+            start (str in ISO 8601 timestamp): When to start the result search.
+            end (str in ISO 8601 timestamp): When to end the result search.
+            range (str): Time range to execute the result search, ending on the current time.
+                Should be in the form "-2w", where y=year, w=week, d=day, h=hour, m=minute, s=second.
+
+        Note:
+            - `range` will take precendent over `start` and `end` if provided.
+
+        Examples:
+            >>> query = api.select(Event).set_time_range(start="2020-10-20T20:34:07Z")
+            >>> second_query = api.select(Event).
+            ...     set_time_range(start="2020-10-20T20:34:07Z", end="2020-10-30T20:34:07Z")
+            >>> third_query = api.select(Event).set_time_range(range='-3d')
+        """
+        args_count = args.__len__()
+        start = kwargs.get("start")
+        end = kwargs.get("end")
+        range = kwargs.get("range")
+
+        if args_count > 0:
+            key = args[0]
+            if key not in ["create_time", "first_event_time", "last_event_time", "last_update_time",
```

Comment:
If we set `backend_timestamp` or `create_time` we should probably also set `_time_range` on top of the criteria as they would need to modify both